sdk/core/azure-core/src/http/telemetry_policy.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <http/policy.hpp>
+
+#include <sstream>
+
+#ifdef _WIN32
+
+#define NOMINMAX
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+namespace {
+
+std::string GetOSVersion()
+{
+  std::ostringstream osVersionInfo;
+
+  {
+    HKEY regKey{};
+    auto regKeyOpened = false;
+    try
+    {
+      if (RegOpenKeyExA(
+              HKEY_LOCAL_MACHINE,
+              "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion",
+              0,
+              KEY_READ,
+              &regKey)
+          == ERROR_SUCCESS)
+      {
+        regKeyOpened = true;
+
+        auto first = true;
+        char const* regValues[]{
+            "ProductName", "CurrentVersion", "CurrentBuildNumber", "BuildLabEx"};
+        for (auto regValue : regValues)
+        {
+          char valueBuf[200] = {};
+          DWORD valueBufSize = sizeof(valueBuf);
+
+          if (RegQueryValueExA(regKey, regValue, NULL, NULL, (LPBYTE)valueBuf, &valueBufSize)
+              == ERROR_SUCCESS)
+          {
+            if (valueBufSize > 0)
+            {
+              osVersionInfo << (first ? "" : " ")
+                            << std::string(valueBuf, valueBuf + (valueBufSize - 1));
+              first = false;
+            }
+          }
+        }
+      }
+
+      RegCloseKey(regKey);
```

Comment:
We only need to close the key if we successfully opened the key.

Reply:
That's what we do - it is inside the `if(OpenKey)`

Reply:
Oh, it wasn't! I'll fix.